### PR TITLE
Generic slice view removal transformation

### PR DESCRIPTION
**Pull request by tbennun**
Supports squeezing and expansion ("unsqueezing"), as well as arbitrary subsets.

### CODE REVIEW

File: dace/sdfg/propagation.py [hunk 1]
```diff
@@ -1202,21 +1202,50 @@ def _propagate_node(dfg_state, node):
...
+
+    # Otherwise, find other data container
+    result = copy.deepcopy(e.data)
+    if dst:
+        node = mpath[-1].dst
+    else:
+        node = mpath[0].src
+
+    # Fix memlet fields
+    result.data = node.data
```
Comment (alexnick83) [inline dace/sdfg/propagation.py:1242]: I am a bit confused about the semantics here. Is it implied that if we reach this part of the code, then the memlet is described (erroneously?) in the exact opposite way compared to the desired "aligned" memlet?

File: dace/transformation/auto/auto_optimize.py [hunk 2]
```diff
@@ -334,7 +334,7 @@ def find_fast_library(device: dtypes.DeviceType) -> List[str]:
         if openblas.OpenBLAS.is_installed():
             result.append('OpenBLAS')
 
-        return result + ['OpenMP', 'pure']
+        return result + ['pure']
```
Comment (alexnick83) [inline dace/transformation/auto/auto_optimize.py:337]: My understanding is that we have an issue with the OpenMP ReduceNode expansion. However, this change will basically disable in the auto-optimizer all OpenMP expansions. We probably don't have many OpenMP expansions now, but perhaps we should amend this in the future.

File: dace/codegen/targets/cpp.py [hunk 3]
```diff
@@ -28,6 +28,9 @@
 from dace.sdfg import SDFG, is_devicelevel_gpu, SDFGState
 from dace.codegen.targets import fpga
 
+if TYPE_CHECKING:
```
[the remark this answers is not in the document]
Comment (tbennun) [reply dace/codegen/targets/cpp.py:31]: this is a standard `typing.TYPE_CHECKING` condition that is used by vscode and pycharm (among others incl. mypy) to use for string imports. This is the recommended Python way :tm:

**Review by alexnick83 — APPROVED** [review]

File: dace/sdfg/propagation.py (hunk 1 above)
Comment (tbennun) [reply dace/sdfg/propagation.py:1242]: I think we should change memlets accordingly

File: dace/transformation/auto/auto_optimize.py (hunk 2 above)
Comment (tbennun) [reply dace/transformation/auto/auto_optimize.py:337]: true. right now the only OpenMP implementation is this reduce node, and openmp is not well supported across compilers :( we had several complaints about this failing.

**Pull request merged**